pkg/asset/installconfig/vsphere/vsphere.go

```diff
@@ -2,10 +2,302 @@
+				Help:    "The Datacenter to be used for installation.",
+			},
+			Validate: survey.Required,
+		},
+	}, selectedDataCenter)
+	selectedDataCenterPath := formatPath(dataCenterPaths[selectedDataCenter])
+	return selectedDataCenter, selectedDataCenterPath, nil
+}
+
+func getCluster(ctx context.Context, path string, finder *find.Finder, client *vim25.Client) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+
+	clusters, err := finder.ClusterComputeResourceList(ctx, path)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to list clusters")
+	}
+
+	if len(clusters) == 0 {
+		return "", errors.New("did not find any clusters")
+	}
+	if len(clusters) == 1 {
+		logrus.Infof("Defaulting to only available cluster: %s", clusters[0].Name())
+		return clusters[0].Name(), nil
+	}
+
+	var clusterChoices []string
+	for _, c := range clusters {
+		clusterChoices = append(clusterChoices, c.Name())
+	}
+	sort.Strings(clusterChoices)
+
+	var selectedcluster string
+	survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Select{
+				Message: "Cluster",
+				Options: clusterChoices,
+				Help:    "The cluster to be used for installation.",
+			},
+			Validate: survey.Required,
+		},
+	}, selectedcluster)
+
+	return selectedcluster, nil
+}
+
+func getDataStore(ctx context.Context, path string, finder *find.Finder, client *vim25.Client) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+
+	dataStores, err := finder.DatastoreList(ctx, path)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to list datastores")
+	}
+
+	if len(dataStores) == 0 {
+		return "", errors.New("did not find any datastores")
+	}
+	if len(dataStores) == 1 {
+		logrus.Infof("Defaulting to only available datastore: %s", dataStores[0].Name())
+		return dataStores[0].Name(), nil
+	}
+
+	var dataStoreChoices []string
+	for _, ds := range dataStores {
+		dataStoreChoices = append(dataStoreChoices, ds.Name())
+	}
+	sort.Strings(dataStoreChoices)
+
+	var selectedDataStore string
+	survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Select{
+				Message: "Default Datastore",
+				Options: dataStoreChoices,
+				Help:    "The default datastore to be used for installation.",
+			},
+			Validate: survey.Required,
+		},
+	}, selectedDataStore)
+	return selectedDataStore, nil
+}
+
+func getNetwork(ctx context.Context, path string, finder *find.Finder, client *vim25.Client) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
+	defer cancel()
+
+	networks, err := finder.NetworkList(ctx, path)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to list networks")
+	}
+
+	if len(networks) == 0 {
+		return "", errors.New("did not find any networks")
+	}
+	if len(networks) == 1 {
+		n := networks[0].(networkNamer)
+		logrus.Infof("Defaulting to only available network: %s", n.Name())
+		return n.Name(), nil
+	}
+
+	var networkChoices []string
+	for _, network := range networks {
```

Comment:
@patrickdillon here I think we would want to check `networks.Reference().Type` and only select:

- DistributedVirtualPortgroup
- Network